AGENTS.md

```diff
@@ -0,0 +1,71 @@
+- Docker Compose for environment and integration test matrices
+- Documentation generation: TypeDoc (API refs) and Docusaurus (site)
+
+## Constraints and Requirements
+
+- Use Node.js ≥20; maintainers develop against Node v24.6.0 (see `.nvmrc`)
+- Prioritize parity with the Python LangChain APIs; propose new abstractions via issues before implementation
+- Optional third-party integrations should live in appropriate `libs/providers/` packages and list optional dependencies in package configs and entrypoint metadata
+
+## Challenges and Mitigation Strategies
+
+- **Environment breadth**: LangChain.js must run across Node ESM/CJS, edge runtimes, and browsers. Rely on environment tests (`pnpm test:exports:docker`) to validate builds across targets.
+- **Third-party dependency surface**: New integrations may require credentials or heavy dependencies. Gate them behind optional peer deps and provide `.env`-driven configuration for integration tests.
+
+## Development Workflow
+
+- Install workspace deps: `pnpm install`
+- Build core packages before depending packages, e.g.:
+  - `cd libs/langchain-core && pnpm install && pnpm build`
+  - Return to repo root for aggregate builds: `pnpm build`
+- Run linting/formatting: `pnpm lint`, `pnpm lint:fix`, `pnpm format`, `pnpm format:check`
+- Run tests:
+  - Unit suite: `pnpm test` (alias for `pnpm test:unit` plus exports tests)
+  - Integration deps up/down: `pnpm test:int:deps`, `pnpm test:int`, `pnpm test:int:deps:down`
+  - Environment/exports matrix (Docker required): `pnpm test:exports:docker`
+  - Dependency range tests (Docker required): `pnpm test:ranges:docker`
+- Watch build utilities: `pnpm --filter @langchain/build watch`
+- Release a package (maintainers): `pnpm release <workspace>` run on a clean `main`
+
+## Coding Guidelines
+
+- Follow TypeScript strictness defined in package `tsconfig` files; prefer composable abstractions aligned with `@langchain/core`
+- Use Prettier (`.prettierrc`) and ESLint (`@langchain/eslint` presets) before submitting PRs
+- Add new public entrypoints in `langchain/langchain.config.js` or `libs/langchain-community/langchain.config.js`, and flag integrations requiring optional deps
+- Place tests alongside source (`tests/*.test.ts` for unit, `*.int.test.ts` for integration)
+
+## Pull Request Guidelines
+
+- Fork-and-PR workflow only; do not push directly to upstream unless you are a maintainer
+- Assign yourself to issues you are addressing and keep scope focused
+- Discuss new abstractions or major API changes in an issue before implementation
+- Run linting, formatting, and relevant tests (unit, integration, exports) before requesting review; note any tests not run and why
+
+## Debugging and Troubleshooting
+
+- Use `nvm use` or similar to ensure Node v24.x; mismatched Node versions often break dependency installation
+- If missing new core export in dependents, rebuild `libs/langchain-core` then rerun dependent build/tests
```

Comment:
Let's generalize this: packages should be build before running any (integration) tests on them